aiohttp/resolver.py

```diff
@@ -37,9 +37,10 @@ async def resolve(
 
         hosts = []
         for family, _, proto, _, address in infos:
-            if family == socket.AF_INET6:
-                if not (socket.has_ipv6 and address[3]):  # type: ignore[misc]
-                    continue
+            if family == socket.AF_INET6 and not socket.has_ipv6:
```

Comment:
`socket.has_ipv6` is a *compile-time* flag.
A host can disable IPv6 support at system startup time also.